lib/active_admin/view_helpers/auto_link_helper.rb

```diff
@@ -21,8 +21,13 @@ def auto_link(resource, content = display_name(resource))
 
       # Like `auto_link`, except that it only returns a URL instead of a full <a> tag
       def auto_url_for(resource)
-        if config = active_admin_resource_for(resource.class)
+        config = active_admin_resource_for(resource.class)
+        return unless config
+
+        if config.controller.action_methods.include?('show') && authorized?(ActiveAdmin::Auth::READ, resource)
```

Comment:
Line is too long. [110/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.